src/storage/knn_index/sparse/bmp_alg.cpp

```diff
@@ -144,8 +148,8 @@ template class TailFwd<f64, i16>;
 template class TailFwd<f64, i8>;
 
 template <typename DataType, typename IdxType>
-Optional<TailFwd<DataType, IdxType>> BlockFwd<DataType, IdxType>::AddDoc(const SparseVecRef<DataType, IdxType> &doc) {
-    SizeT tail_size = tail_fwd_.AddDoc(doc);
+Optional<TailFwd<DataType, IdxType>> BlockFwd<DataType, IdxType>::AddDoc(const SparseVecRef<DataType, IdxType> &doc, SizeT &mem_usage) {
+    SizeT tail_size = tail_fwd_.AddDoc(doc, mem_usage);
```

Comment:
Do not add mem_usage in tail_fwd, because in line 157, tail fwd is swap and data is transfered to block_terms.